wfe/wfe.go

```diff
@@ -1174,6 +1175,40 @@ func (wfe *WebFrontEndImpl) Authorization(ctx context.Context, logEvent *request
 		return
 	}
 
+	if wfe.AllowAuthzDeactivation && request.Method == "POST" {
+		body, _, reg, prob := wfe.verifyPOST(ctx, logEvent, request, true, core.ResourceAuthz)
+		addRequesterHeader(response, logEvent.Requester)
+		if prob != nil {
+			wfe.sendError(response, logEvent, prob, nil)
+			return
+		}
+		if reg.ID != authz.RegistrationID {
+			logEvent.AddError("registration ID doesn't match ID for authorization")
+			wfe.sendError(response, logEvent, probs.Unauthorized("Registration ID doesn't match ID for authorization"), nil)
+			return
+		}
+		var req struct {
+			Status core.AcmeStatus
+		}
+		err = json.Unmarshal(body, &req)
+		if err != nil {
+			wfe.sendError(response, logEvent, probs.Malformed("Error unmarshaling JSON"), err)
+			return
+		}
+		if req.Status != core.StatusDeactivated {
+			logEvent.AddError("invalid status value")
+			wfe.sendError(response, logEvent, probs.Malformed("Invalid status value"), err)
+			return
+		}
+		err = wfe.RA.DeactivateAuthorization(ctx, authz)
+		if err != nil {
+			logEvent.AddError("unable to deactivate authorization", err)
+			wfe.sendError(response, logEvent, probs.Malformed("Error deactivating authorization"), err)
+			return
+		}
+		authz.Status = core.StatusDeactivated
```

Comment:
It is done in `ra.DeactivateAuthorization` but since we are a. passing by value and b. crossing a RPC boundary (so passing by reference wouldn't actually mutate the object) the passed in authorization isn't changed in this scope.

This is a little hacky way to avoid having to do another `GetAuthorization` call in order to present the deactivated authorization to the user, but I could probably be convinced it isn't necessary.